packages/js/product-editor/src/components/attribute-control/new-attribute-modal.tsx

```diff
@@ -238,6 +230,16 @@ export const NewAttributeModal: React.FC< NewAttributeModalProps > = ( {
 					// eslint-disable-next-line @typescript-eslint/no-explicit-any
 					setValue: ( name: string, value: any ) => void;
 				} ) => {
+					/*
+					 * Handle if the Add button is disabled.
+					 * - attributes should be not empty array
+					 * - attributes should not be an array of null or undefined
+					 * - It should be at least one term or option in the attribute
+					 */
+					const isAddButtonDisabled = ! values.attributes.some(
+						( attr ) => isAttributeFilledOut( attr )
+					);
```

Comment:
That's acceptable to me.
The user learned the difference between the accepted and unaccepted terms. Visually, the difference is more than clear.

<img width="129" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/1f721283-dc74-45ad-9f1d-eda77617ec8b">

Also, your example allows the creation of at least one variation.
I think allowing the user to create the variations in that state makes sense.

Reply:
I disagree. From my perspective as a user... "Whether I have learned something of not in a system, the reality is that I'm using multiple systems/apps throughout my day, I'm busy and distracted, and sometimes I just make a mistake. I appreciate a system that doesn't leave me wondering why my change didn't take, when I turned out I just forgot to hit the Enter key."

Reply:
okay. let's go for your perspective. Let's remove the comments block, too

Reply:
It doesn't seem to be simple to address:
* We need to register if there is an unaccepted instance for each token field component (we can use [onInputChange](https://github.com/WordPress/gutenberg/blob/1533a4213413cb7856748bcfeaa856324538796e/packages/components/src/form-token-field/index.tsx#L59) for that)
* Check that all token field instances don't have unaccepted terms.

It will require more significant changes in the code. Can we handle this in a follow-up?

Reply:
Okay, let's do that.

Reply:
Oh wait... re-testing it looks like you were able to fix this? Need to look at code.